Stripe/StripeiOSTests/STPPaymentHandlerFunctionalTest.swift

```diff
@@ -190,4 +191,156 @@ final class STPPaymentHandlerFunctionalSwiftTest: XCTestCase, STPAuthenticationC
         }
         self.waitForExpectations(timeout: 10)
     }
+
+    // MARK: - Test payment handler sends analytics
+
+    func test_confirm_payment_intent_sends_analytic() {
+        // Confirming a hardcoded already-confirmed PI with invalid params...
+        let paymentIntentParams = STPPaymentIntentParams(clientSecret: "pi_3P20wFFY0qyl6XeW0dSOQ6W7_secret_9V8GkrCOt1MEW8SBmAaGnmT6A", paymentMethodType: .card)
+        let paymentHandlerExpectation = expectation(description: "paymentHandlerExpectation")
+        let paymentHandler = STPPaymentHandler(apiClient: STPAPIClient(publishableKey: STPTestingDefaultPublishableKey))
+        let analyticsClient = STPAnalyticsClient()
+        paymentHandler.analyticsClient = analyticsClient
+        paymentHandler.confirmPayment(paymentIntentParams, with: self) { (_, _, _) in
+            // ...should send these analytics
+            paymentHandlerExpectation.fulfill()
+            let firstAnalytic = analyticsClient._testLogHistory.first
+            XCTAssertEqual(firstAnalytic?["event"] as? String, STPAnalyticEvent.paymentHandlerConfirmStarted.rawValue)
+            XCTAssertEqual(firstAnalytic?["intent_id"] as? String, "pi_3P20wFFY0qyl6XeW0dSOQ6W7")
+            XCTAssertEqual(firstAnalytic?["payment_method_type"] as? String, "card")
+            let lastAnalytic = analyticsClient._testLogHistory.last
+            XCTAssertEqual(lastAnalytic?["event"] as? String, STPAnalyticEvent.paymentHandlerConfirmFinished.rawValue)
+            XCTAssertEqual(lastAnalytic?["intent_id"] as? String, "pi_3P20wFFY0qyl6XeW0dSOQ6W7")
+            XCTAssertEqual(lastAnalytic?["status"] as? String, "failed")
+            XCTAssertEqual(lastAnalytic?["payment_method_type"] as? String, "card")
+            XCTAssertEqual(lastAnalytic?["error_type"] as? String, "invalid_request_error")
+            XCTAssertEqual(lastAnalytic?["error_code"] as? String, "payment_intent_unexpected_state")
```

Comment:
Does any of this code get ran? The expectation is fulfilled before, so I'm guessing the `waitForExpectations` finishes before these asserts are ran.

Reply:
Same comment for the other tests.

Reply:
It does in practice but it can't hurt to put it at the end.